### PR TITLE
Va iir 688 remove v0 controller and associated models

### PR DESCRIPTION
**Pull request by AdamKing0126**
## Summary

Dead Code Removal

I'm Adam King, I work on the VA-IIR team and we own this component.

The GI Bill Statement of Benefits tool previously made a network request to the EVSS service, which is slated to be decommissioned in 2024.  A new API endpoint at `v1/post911_gi_bill_status` was created to replace `v0/post911_gi_bill_status` and request data from Lighthouse.  The new endpoint has been active in production for some time, and the network call to the old `v0` endpoint has been removed from the `vets-website` application.  There is no traffic flowing to this endpoint any longer ([Datadog Dashboard](https://vagov.ddog-gov.com/dashboard/kpe-8jk-rik/post-911-gi-bill-statuses-controller?fromUser=false&refresh_mode=sliding&view=spans&from_ts=1717176521486&to_ts=1717180121486&live=true)). This PR represents the last step in the removal of the old code, and the decommissioning of the `v0/post911_gi_bill_status` endpoint.

Doing this required the removal of a significant amount of code, so the removal was broken up into four PRs, with this being the final PR.  This PR removes the `v0` controller, associated models, and the entry in `config/routes.rb`.

Previous work:
- [16944](https://github.com/department-of-veterans-affairs/vets-api/pull/16944): ~remove controller tests~ MERGED
- [16945](https://github.com/department-of-veterans-affairs/vets-api/pull/16945): ~remove response and service tests~ MERGED
- [16953](https://github.com/department-of-veterans-affairs/vets-api/pull/16953): ~final test cleanup and swagger tests~ MERGED

## Related issue(s)

[GitHub Issue](https://github.com/department-of-veterans-affairs/va-iir/issues/688)

## Screenshots

Screenshot 2024-05-31 at 1 29 52 PM

## What areas of the site does it impact?

This PR removes dead code which is currently unreachable.

## Acceptance criteria

- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Swagger Documentation has not yet been updated. [16891](https://github.com/department-of-veterans-affairs/vets-api/pull/16891) will be put up for code review once this PR is merged
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -856,8 +855,7 @@ lib/evss/disability_compensation_auth_headers.rb @department-of-veterans-affairs
 lib/evss/disability_compensation_form @department-of-veterans-affairs/Disability-Experience @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 lib/evss/documents_service.rb @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 lib/evss/error_middleware.rb @department-of-veterans-affairs/vfs-authenticated-experience-backend @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
-lib/evss/gi_bill_status @department-of-veterans-affairs/my-education-benefits @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
-lib/evss/intent_to_file @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
+lib/evss/intent_to_file @department-of-veterans-affairs/Benefits-Team-1 @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
```
Comment (stevelong00) [inline .github/CODEOWNERS:858]: Why was the benefits team added to intent_to_file?

Comment (AdamKing0126) [reply .github/CODEOWNERS]: That's really weird.  I have no explanation of how that could've happened.  I removed that change.

**Review by stevelong00 — APPROVED** [review]
Looks great!

**Comment by AdamKing0126** [comment]
@department-of-veterans-affairs/backend-review-group - I'm going to need a "Danger exception" for this.  Couldn't get the PR under 500 lines

**Review by ericboehs — APPROVED** [review]

**Pull request merged**